two-factor.php

```diff
@@ -50,54 +24,44 @@ function wpcom_vip_force_two_factor() {
 		return false;
 	}
 
-	return apply_filters( 'wpcom_vip_force_two_factor', false );
+	// We can't use current_user_can because it calls map_meta_cap which creates an infinite loop
+	$user = wp_get_current_user();
+	$caps = array_keys( $user->allcaps );
+	return apply_filters( 'wpcom_vip_force_two_factor', in_array( 'edit_posts', $caps ) );
```

Comment:
This is a bit of a hack because we can't use `current_user_can()`. Also, almost every user has `edit_posts`, including contributors. It might make more sense to default this to on for everyone?

Reply:
I think for releasing this, we should look at updating the admin notice notifying that this will be enforced at `x` date with clear links to enabling 2fa.

Reply:
> It might make more sense to default this to on for everyone?

This would impact non-editorial users on WooCommerce and other sites (e.g. sites with subscriptions or other community features), which is probably not ideal.

Reply:
Also, maybe doing this on-the-fly every time, we should make the determination around whether to enforce 2fa on the `set_current_user` filter and store as a `static` somewhere? We could then check that rather than relying on the hack.

Reply:
Fixed in https://github.com/Automattic/vip-go-mu-plugins/pull/1213/commits/c66a3df50d5a96d7c7e67787ac13f796c6c798c2